Version v11.7.2

danjm opened the pull request.

Gudahtt commented on CHANGELOG.md:
```diff
@@ -6,12 +6,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
+## [11.7.2]
+### Added
```
This changelog needs to be fixed; we have two empty sections and we're missing a change entry

DDDDDanica replied:
Hey @Gudahtt i believe this list is still WIP but i have included the latest changes here [#1a6d367a](https://github.com/MetaMask/metamask-extension/pull/22336/commits/1a6d367a298eff2ea6d18381c2cbc74b82fced12)  please take a look at it and let me know if there's any issues.

Gudahtt commented on privacy-snapshot.json:
```diff
@@ -38,5 +38,6 @@
   "unresponsive-rpc.url",
-  "www.4byte.directory"
+  "www.4byte.directory",
+  "price-api.metafi.codefi.network"
```
We should be able to remove `api.coingecko.com` from this list now as well

danjm replied:
Fixed here https://github.com/MetaMask/metamask-extension/pull/22372

Gudahtt commented on app/scripts/metamask-controller.js:
```diff
@@ -571,7 +565,7 @@ export default class MetamaskController extends EventEmitter {
-      onNetworkStateChange: networkControllerMessenger.subscribe.bind(
+      onNetworkDidChange: networkControllerMessenger.subscribe.bind(
         networkControllerMessenger,
         'NetworkController:stateChange',
```
We should be subscribed to `networkDidChange` here

danjm replied:
@Gudahtt The code changes you commented on are now part of v11.7.3 https://github.com/MetaMask/metamask-extension/pull/22354

I wanted to clarify on the point you made here though. Are you saying these lines of code should be changed to:
```
onNetworkDidChange: networkControllerMessenger.subscribe.bind(
    networkControllerMessenger,
    'NetworkController:networkDidChange',
)
```

danjm replied:
Fixed here https://github.com/MetaMask/metamask-extension/pull/22372

Gudahtt commented on app/scripts/metamask-controller.js:
```diff
@@ -571,7 +565,7 @@ export default class MetamaskController extends EventEmitter {
+      onNetworkDidChange: networkControllerMessenger.subscribe.bind(
         networkControllerMessenger,
         'NetworkController:stateChange',
       ),
```
A few lines below here, the `getERC20TokenName` option should be removed. This option is not accepted as of v21

>BREAKING: TokensController.watchAsset now performs on-chain validation of the asset's symbol and decimals, if they're defined in the contract (https://github.com/MetaMask/core/pull/1745)
>
>    The TokensController constructor no longer accepts a getERC20TokenName option. It was no longer needed due to this change.

danjm replied:
Fixed here https://github.com/MetaMask/metamask-extension/pull/22372

Gudahtt commented:
I found a few minor issues with the code changes here, but we should fix them on `develop` first before fixing them here. Any changes introduced in this PR will be undone with the sync back to `develop`.

matthewwalsh0 committed:
Fix incorrect transaction statuses (#22319)

Fix the `dropped` status detection.
Ensure previously pending transactions are monitored after switching
network and not stuck as `pending`.
Prevent `failed` transactions on networks that use pending transaction
receipts with `null` status.
Fix recording of `v` value.
Fix usage of saved gas fees.

danjm committed:
v11.7.2

danjm commented:
The commits in this PR have changed from when it was first cut. What was v11.7.2 yesterday is now v11.7.3 https://github.com/MetaMask/metamask-extension/pull/22354

This new v11.7.2 contains a single commit focused on fixing transaction status display bugs

sleepytanya commented:
Confirmed for all test networks on Chrome and Firefox:

1. incoming transactions history on Linea Goerli is fixed https://github.com/MetaMask/metamask-extension/issues/22237
2. saved gas fee (Advanced gas) applied to all subsequent transactions unless user changes gas settings to another option https://github.com/MetaMask/metamask-extension/issues/22333
3. I don't see 'dropped' transaction issue (scenario 1 - create tx, switch network, create tx with the same account and with the same nonce, scenario 2 - send eth between two accounts on the same wallet and same network with the same nonce / higher / lower nonce) https://github.com/MetaMask/metamask-extension/issues/22312
4. I sent around 50 txs while testing but encountered 'pending' transaction only once and the status was updated pretty quick so we can count it as accidental one-time issue (?) and consider 'pending' transaction issue fixed https://github.com/MetaMask/metamask-extension/issues/21941
